components/empty-state/empty-state-illustrated-link.js

```diff
@@ -1,62 +1,127 @@
 import '../link/link.js';
-import { bodyCompactStyles, heading2Styles  } from '../typography/styles.js';
 import { emptyStateIllustratedStyles, emptyStateStyles } from './empty-state-styles.js';
-import { html, LitElement } from 'lit';
-import { getIllustration } from './getIllustration.js';
+import { html, LitElement, nothing } from 'lit';
+import { bodyCompactStyles } from '../typography/styles.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { loadSvg } from '../../generated/empty-state/presetIllustrationLoader.js';
 import { RtlMixin } from '../../mixins/rtl-mixin.js';
 import { runAsync } from '../../directives/run-async/run-async.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
 
 /**
  * The `d2l-empty-state-illustrated-link` component is an empty state component that displays an illustration and action link. The illustration property can be set to use one of the preset illustrations or a custom SVG illustration can be added in the default slot.
- * @slot - Custom SVG content if illustration property is not set
+ * @slot - Custom SVG content if `illustration-name` property is not set
  */
 class EmptyStateIllustratedLink extends RtlMixin(LitElement) {
 
 	static get properties() {
 		return {
 			/**
 			  * Optional: The action URL or URL fragment of the link
-			  * @type {String}
+			  * @type {string}
 			  */
 			actionHref: { type: String, attribute: 'action-href' },
 			/**
 			 * Optional: The action text to be used in the link
-			 * @type {String}
+			 * @type {string}
 			 */
 			actionText: { type: String, attribute: 'action-text' },
 			/**
 			 * REQUIRED: A description giving details about the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
 			description: { type: String },
 			/**
 			 * Optional: The name of the preset image you would like to display in the component
-			 * @type {String}
+			 * @type {string}
 			 */
-			illustration: { type: String },
+			illustrationName: { type: String, attribute: 'illustration-name' },
 			/**
 			 * REQUIRED: A title for the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
-			title: { type: String },
+			titleText: { type: String, attribute: 'title-text' },
+			_contentWidth: { type: Number }
 		};
 	}
 
 	static get styles() {
-		return [emptyStateStyles, emptyStateIllustratedStyles, bodyCompactStyles, heading2Styles];
+		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
+	}
+
+	constructor() {
+		super();
+
+		this._contentWidth = 700;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
```

Comment:
These are now sharing a lot more logic than they originally were... wondering if it might make sense to move that duplicated stuff into a base class, similar to what @eKoopmans did with the [object property items](https://github.com/BrightspaceUI/core/blob/main/components/object-property-list/object-property-list-item-link.js#L10).

Reply:
Yep, they are very similar apart from the specific actions that they have. I think it would make sense